src/components/NodeLink/functionUpdateVis.js

```diff
@@ -175,11 +175,13 @@ function makeSimulation() {
   return simulation;
 }
 
-function nodeFill(node) {
-  const { attributes, colorClasses, nodeColors } = this;
-  if (attributes.nodeFill === "table") {
-    const index = colorClasses.findIndex(d => { return d === node.id.split("/")[0] }) % 5
+function nodeFill(node, nodeClasses) {
+  const { colorClasses, nodeColors, colorVariable } = this;
+  if (colorVariable === "table") {
+    const index = colorClasses.findIndex(d => { return d === node.id.split("/")[0] }) % nodeColors.length
     return nodeColors[index]
+  } else {
+    return nodeColors[nodeClasses.findIndex(d => { return d === node[colorVariable]}) % nodeColors.length]
```

Comment:
Instead of these linear searches, why not just use d3 colormapping utilities?

Reply:
Should be fixed in 63e88a2